react/week2/todolist/src/components/TodoLists.js

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from "react";
+import Todo from "./Todo";
+import TodoForm from "./TodoForm";
+
+function TodoLists() {
+  const [todos, setTodos] = useState([]);
+
+  const addTodo = (todo) => {
+    // utilize the input
+    if (!todo.text || /^\s*$/.test(todo.text)) {
+      return;
+    }
+
+    const newTodos = [todo, ...todos];
+
+    setTodos(newTodos);
+  };
+
+  const updateTodo = (todoId, newValue) => {
+    if (!newValue.text || /^\s*$/.test(newValue.text)) {
+      return;
+    }
+
+    setTodos((prev) =>
+      prev.map((item) => (item.id === todoId ? newValue : item))
+    );
+  };
+
+  const removeTodo = (id) => {
+    const removeArr = [...todos].filter((todo) => todo.id !== id);
+
+    setTodos(removeArr);
+  };
+
+  const completeTodo = (id) => {
+    let updatedTodos = todos.map((todo) => {
+      if (todo.id === id) {
+        todo.isComplete = !todo.isComplete;
+      }
+      return todo;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  return (
+    <div>
+      <h1>What's the Plan for Today?</h1>
+      <TodoForm onSubmit={addTodo} />
+      <Todo
+        todos={todos}
+        completeTodo={completeTodo}
+        removeTodo={removeTodo}
+        updateTodo={updateTodo}
+      />

```

Comment:
Good job defining your functions here and passing them on! Also, you created a separate component for adding a todo, well done!
However, since this component is your TodoList, it'd be better if you iterated over the todos and did 
```
{todos.map((todo) => {
        return <Todo todo={todo}
                     completeTodo={completeTodo}
                     removeTodo={removeTodo}
                     updateTodo={updateTodo} />;
      })}
```

Reply:
I changes as you advised, and this is shown.
![image](https://user-images.githubusercontent.com/84173609/143595957-44d65f5e-7363-4363-85d0-d9ee33cad63b.png)

since I do mapping on the todo, it doesn't like that.
should I change to mapping only in there(todolist)?